src/mainboard/protectli/vault_adl_n/variants/vp2430/overridetree.cb

```diff
@@ -0,0 +1,99 @@
+chip soc/intel/alderlake
+	# FSP configuration
+
+	register "common_soc_config" = "{
+		.i2c[0] = {
+			.speed = I2C_SPEED_FAST,
+			.rise_time_ns = 80,
+			.fall_time_ns = 110,
+		},
+	}"
+
+	device domain 0 on
+		device ref igpu on
+			register "ddi_portA_config" = "1" # HDMI on port A
+			register "ddi_portB_config" = "1" # DP on port B
+			register "ddi_ports_config" = "{
+				[DDI_PORT_A] = DDI_ENABLE_HPD | DDI_ENABLE_DDC,
+				[DDI_PORT_B] = DDI_ENABLE_HPD | DDI_ENABLE_DDC,
+				[DDI_PORT_1] = DDI_ENABLE_HPD,
+				[DDI_PORT_2] = DDI_ENABLE_HPD,
+			}"
+		end
+
+		device ref sata on
+			register "sata_ports_enable[0]" = "1"
+		end
+
+		# LAN1
+		device ref pcie_rp2 on
+			register "pch_pcie_rp[PCH_RP(2)]" = "{
+				.flags = PCIE_RP_LTR | PCIE_RP_AER | PCIE_RP_CLK_REQ_UNUSED,
+				.clk_src = 0,
+			}"
+		end
+		# LAN2
+		device ref pcie_rp3 on
+			register "pch_pcie_rp[PCH_RP(3)]" = "{
+				.flags = PCIE_RP_LTR | PCIE_RP_AER | PCIE_RP_CLK_REQ_UNUSED,
+				.clk_src = 1,
+			}"
+		end
+		# LAN 3
+		device ref pcie_rp4 on
+			register "pch_pcie_rp[PCH_RP(4)]" = "{
+				.flags = PCIE_RP_LTR | PCIE_RP_AER | PCIE_RP_CLK_REQ_UNUSED,
+				.clk_src = 2,
+			}"
+		end
+		# LAN4
+		device ref pcie_rp7 on
+			register "pch_pcie_rp[PCH_RP(7)]" = "{
+				.flags = PCIE_RP_LTR | PCIE_RP_AER | PCIE_RP_CLK_REQ_UNUSED,
+				.clk_src = 3,
+			}"
+		end
+		# WIFI
+		device ref pcie_rp12 on
+			register "pch_pcie_rp[PCH_RP(12)]" = "{
+				.flags = PCIE_RP_LTR | PCIE_RP_AER | PCIE_RP_CLK_REQ_UNUSED,
+				.clk_src = 4,
+			}"
+		end
+		# NVMe x2 link
+		device ref pcie_rp9 on
+			register "pch_pcie_rp[PCH_RP(9)]" = "{
+				.flags = PCIE_RP_LTR | PCIE_RP_AER | PCIE_RP_CLK_REQ_UNUSED,
+				.clk_src = 0,
+				}"
+			smbios_slot_desc	"SlotTypeM2Socket3" "SlotLengthOther"
+						"M.2/M 2280 (M2_NVME2X)" "SlotDataBusWidth2X"
+		end
+
+		device ref cnvi_wifi on
+			register "cnvi_bt_core" = "true"
+			chip drivers/wifi/generic
+				# TODO how to figure out the wake
+				register "wake" = "GPE0_PME_B0"
+				device generic 0 on end
+			end
+		end
+
+		device ref i2c0 on
+			register "serial_io_i2c_mode[PchSerialIoIndexI2C0]" = "PchSerialIoPci"
+		end
+
+		device ref i2c4 on
+			register "serial_io_i2c_mode[PchSerialIoIndexI2C4]" = "PchSerialIoPci"
+			chip drivers/i2c/generic
+				register "hid" = ""INT3515""
+				register "uid" = "1"
+				register "name" = ""PD01""
+				register "desc" = ""TPS65994 USB-C PD Controller""
+				register "irq_gpio" = "ACPI_GPIO_IRQ_LEVEL_LOW(GPP_F13)"
+				# TODO fix device number
+				device i2c 23 on end
```

Comment:
There seem to be two addresses for 2 USb-C ports: 0x23 and 0x27. This will have to be added somehow. Not sure yet how.

Reply:
Linux expects multiple ACPI SerialBus resources

https://github.com/torvalds/linux/blob/906bd684e4b1e517dd424a354744c5b0aebef8af/drivers/acpi/scan.c#L1755-L1774

Reply:
Exactly, each device must be reported in CRS, but coreboot probably can't do it with generic i2c driver I think...

Reply:
https://github.com/Dasharo/coreboot/pull/576/commits/d25d148fd1f1de1129b31005f821f5b94ca00354

Added custom FSP with the missing parameters.

Reply:
Unified the I2C config and USB-PD: https://github.com/Dasharo/coreboot/pull/576/commits/ac64c10e657671a379e0a19b53f153a7e7911454

It seems the older schematics for VP32xx make more sense than the newer one. Newer schematics does not have anything that could driver USB-C CC pins (seems like the parts were purposefully removed from the file).

When looked at the older schematics the I2C and USB-PD is the same. Added the ASL source for PD manually including all the I2C device addresses.